### PR TITLE
ADEN-13247 - Moved directedAtChildren out of Blockthrough scope

### PR DESCRIPTION
**Pull request by domjanik**
In this PR I've moved directedAtChildren flag out of Blockthrough scope. I did that, because we found out that it is being used by other partners as well and its better to keep such values out of single partners scope.

### CODE REVIEW

File: spec/ad-services/bt-rec/bt-rec.spec.ts [hunk 1]
```diff
@@ -58,15 +58,15 @@ describe('BlockThrough recovery', () => {
 		loadScriptStub.returns(Promise.resolve());
 		context.set('wiki.targeting.directedAtChildren', true);
 		await btRec.run().then(() => {
-			expect(window.fandomContext.partners.blockthrough.directedAtChildren).to.equal(true);
```
Comment (jbienkowski-fandom) [inline spec/ad-services/bt-rec/bt-rec.spec.ts:61]: are we not interested in backwards compatibility?

File: src/ad-services/bt-rec/index.ts [hunk 2]
```diff
@@ -20,9 +20,7 @@ class BTRec {
 
 		utils.logger(logGroup, 'loading');
 		globalContextService.setValue(GlobalContextCategories.partners, {
```
Comment (szymon-nawrocki) [inline src/ad-services/bt-rec/index.ts:22]: This code sets up `partners.directedAtChildren` at `fandomContext` object and now is not only BTRec related.
I think it should be moved out of `/ad-services/bt-rec` catalogue

File: spec/ad-services/bt-rec/bt-rec.spec.ts (hunk 1 above)
Comment (domjanik) [reply spec/ad-services/bt-rec/bt-rec.spec.ts:61]: We're going to inform our partners about that and sync it with them. I think we do not need such compatibility right now 🤔

File: src/core/utils/is-coppa-subject.ts [hunk 3]
```diff
@@ -7,12 +7,25 @@ export function isCoppaSubject(): boolean {
 		try {
 			const cookieStorage = new CookieStorageAdapter();
 			const ageGateResult = cookieStorage.getItem('ag');
+			const isCoppaSubject = ageGateResult === '1' ? false : wikiDirectedAtChildren;
 
-			return ageGateResult === '1' ? false : wikiDirectedAtChildren;
+			globalContextService.setValue(GlobalContextCategories.partners, {
+				directedAtChildren: !!isCoppaSubject,
+				blockthrough: {
```
Comment (jbienkowski-fandom) [inline src/core/utils/is-coppa-subject.ts:14]: backward compatible after all? :) let `@types/fandom/index.d.ts` reflect that change too please

File: src/core/utils/is-coppa-subject.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import { utils } from '../index';
+import { GlobalContextCategories, globalContextService, utils } from '../index';
 import { context, CookieStorageAdapter } from '../services';
 
 export function isCoppaSubject(): boolean {
```
Comment (jbienkowski-fandom) [inline src/core/utils/is-coppa-subject.ts:4]: we should check `globalContextService.getValue` and use it as "cache" as `utils.isCoppaSubject` is used quite a lot

**Review by jbienkowski-fandom — CHANGES_REQUESTED** [review]

File: src/core/utils/is-coppa-subject.ts (hunk 3 above)
Comment (domjanik) [reply src/core/utils/is-coppa-subject.ts:14]: Yeah... I thought about this and after all I'm kinda afraid of letting this only up to our partners xD

File: src/core/utils/is-coppa-subject.ts (hunk 4 above)
Comment (szymon-nawrocki) [inline src/core/utils/is-coppa-subject.ts:4]: Also this logic is duplicated in IdentityEngine, maybe this is a good moment to unify it between the repos?
https://github.com/Wikia/identity-engine/blob/main/src/modules/shared/CoppaSubject/index.ts

**Review by szymon-nawrocki — APPROVED** [review]

**Review by jbienkowski-fandom — DISMISSED** [review]

**Review by szymon-nawrocki — DISMISSED** [review]

File: src/core/services/global-context-service.ts [hunk 5]
```diff
@@ -62,6 +65,9 @@ export class GlobalContextService {
 		category: K,
 		key?: I,
 	): WindowFandomContext[K][I] | undefined {
+		if (!window.fandomContext) {
+			return;
```
Comment (jbienkowski-fandom) [inline src/core/services/global-context-service.ts:69]:
```suggestion
			return undefined;
```

File: src/core/services/global-context-service.ts [hunk 6]
```diff
@@ -62,6 +65,9 @@ export class GlobalContextService {
...
+		}
 		const categoryObj = window.fandomContext[category];
 		if (categoryObj) {
 			return categoryObj[key];
```
Comment (jbienkowski-fandom) [inline src/core/services/global-context-service.ts:73]: (nit1) the `else` statement is unnecessary, so let's get rid of it
(nit2) `return undefined;` at the end (explicit is better than implicit)

**Review by szymon-nawrocki — APPROVED** [review]

**Review by jbienkowski-fandom — APPROVED** [review]
LGTM

**Pull request merged**